File: judge-web/src/core/repository/dto/response/LeaderboardOutputDTO.ts
export type LeaderboardOutputDTO = {
  contestId: number;
  problems: {
    id: number;
    title: string;
  }[];
  members: {
    id: number;
    name: string;
    problems: {
      id: number;
      wrongSubmissions: number;
      isAccepted: boolean;
      penalty: number;
    }[];
    score: number;
    penalty: number;
  }[];
};

File: judge-backend/common/src/main/kotlin/io/github/leonfoliveira/judge/common/service/dto/output/LeaderboardOutputDTO.kt
package io.github.leonfoliveira.judge.common.service.dto.output

import java.time.OffsetDateTime
import java.util.UUID

data class LeaderboardOutputDTO(
    val contestId: UUID,
    val slug: String,
    val startAt: OffsetDateTime,
    val members: List<MemberDTO>,
    val issuedAt: OffsetDateTime,
) {
    data class MemberDTO(
        val id: UUID,
        val name: String,
        val score: Int,
        val penalty: Int,
        val problems: List<ProblemDTO>,
    )

    data class ProblemDTO(
        val id: UUID,
        val letter: Char,
        val isAccepted: Boolean,
        val acceptedAt: OffsetDateTime? = null,
        val wrongSubmissions: Int,
        val penalty: Int,
    )
}
